lib/ports.dart

```diff
@@ -87,25 +87,25 @@ SendPort singleCallbackPort(void callback(response),
 /// completed in response to another event, either a port message or a timer.
 ///
 /// Returns the `SendPort` expecting the single message.
-SendPort singleCompletePort(Completer completer,
-    {callback(message), Duration timeout, onTimeout()}) {
+SendPort singleCompletePort<T>(Completer completer,
+    {callback(T message), Duration timeout, T onTimeout()}) {
   if (callback == null && timeout == null) {
     return singleCallbackPort(completer.complete);
   }
   RawReceivePort responsePort = new RawReceivePort();
-  var timer;
+  Timer timer;
   if (callback == null) {
-    responsePort.handler = (response) {
+    responsePort.handler = (T response) {
```

Comment:
I was surprised that this assignment is valid.
It's because `RawReceivePort.handler` has type `Function`. 
I guess we'll want to change that at some point, not sure if to `void Function(Object)` or `void Function(Null)`. The former is safe, the latter is practical. Pick one.

For now, I think I'd prefer to keep `response` as `Object` or `dynamic` and doing a cast below where you use it.

Reply:
We can't use the new Function syntax ... unless we want to punt on 1.x support for this package. Will fix, though...